rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIT.java

```diff
@@ -0,0 +1,497 @@
+package co.rsk.peg.feeperkb;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.InMemoryStorage;
+import co.rsk.peg.feeperkb.constants.FeePerKbConstants;
+import co.rsk.peg.feeperkb.constants.FeePerKbMainNetConstants;
+import co.rsk.peg.storage.StorageAccessor;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(Lifecycle.PER_CLASS)
+class FeePerKbIT {
+    private static final Coin differentFeePerKbVote = Coin.valueOf(60_000L);
+    private final FeePerKbConstants feePerKbConstants = FeePerKbMainNetConstants.getInstance();
+    private FeePerKbSupport feePerKbSupport;
+    private SignatureCache signatureCache;
+    private Coin currentFeePerKb;
+
+    @BeforeAll
+    void setUp() {
+        StorageAccessor inMemoryStorage = new InMemoryStorage();
+        FeePerKbStorageProvider feePerKbStorageProvider = new FeePerKbStorageProviderImpl(inMemoryStorage);
+        feePerKbSupport = new FeePerKbSupportImpl(feePerKbConstants, feePerKbStorageProvider);
+        signatureCache = mock(SignatureCache.class);
+    }
+
+    @Test
+    @Order(0)
+    void getGenesisFeePerKb_whenNotExistsPreviousVoting_shouldReturnGenesisFeePerKbValue() {
+        Coin genesisFeePerKb = feePerKbConstants.getGenesisFeePerKb();
+        currentFeePerKb = genesisFeePerKb;
```

Comment:
shouldn't we access the `currentFeePerKb` value through `getFeePerKb()` method?

Reply:
or at least set it after the assertion

Reply:
We use this value in the `assertFeePerKbValue` method. It stands for the expected `currentFeePerKb`, and we compare it with `getFeePerKb()` to validate we are getting the correct current value.

About the position of this line of code, it is set in all the methods after the current Fee per Kb is set. Here corresponds after call `genesisFeePerKb()`, which initiates the current fee per KB value.

Reply:
pretty sure you are not using the `currentFeePerKb` value in the assertion...
You are passing the `genesisFeePerKb` [as an input](https://github.com/rsksmart/rskj/pull/2433/files/f4e06f289f974c733e66c139fcabee30e240c30f#diff-72f83e7f5b9745bce799c9de61da92d8ffb07cd4646ef1da1758d11431208939R45) and then comparing it to the one [being obtained from the `getFeePerKb`](https://github.com/rsksmart/rskj/blob/f4e06f289f974c733e66c139fcabee30e240c30f/rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIT.java#L483) method.

Reply:
after asserting that the value obtained from the method is the genesis one, it makes sense to set
`currentFeePerKb = genesisFeePerKb`.

Reply:
Yes, that's correct! In each method, you can see the `assertFeePerKbValue` method is used a couple of times or more with different inputs such as `currentFeePerKb` or the value voted by the majority of authorizers. 

**Context:** 
When you use it the first time, we want to ensure the current fee per kb doesn't change its value because it doesn't have to change in the first moment (for this case it doesn't apply since the first voting doesn't happen yet. In this first case, it is used to guarantee it has the value of the previous voting round, for this reason, we used `currentFeePerKb` as an input parameter for this case). On the other hand, in the case when the `assertFeePerKbValue` method is used at the end of methods, we want to ensure the current fee per kb has changed correctly and it has the value voted by the majority of the authorizers. 
In this case, we assume `genesisFeePerKb` is a "kind of value voted", and we use the last approach explained.

Reply:
About changing the position of the line indicated, I wouldn't put at the end of the test that line. I'd choose to use assertions to keep using the `AAA` pattern in the tests.